core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -1418,7 +1418,22 @@ else if (res.operationResult == RETRY) {
             if (value == null) {
                 return null;
             }
-            return new AbstractMap.SimpleImmutableEntry<>(key, value);
+            return this;
```

Comment:
I would prefer a more comprehensive solution :) Like enhancement of the IterItem to extend Map.Entry... So it can be also used in the future cases...

But this works as well :)